src/background/executor.test.ts

```
@@ -42,17 +42,17 @@ const runBrickMock = jest.mocked(runBrick);
 
 const optionsFactory = define<RemoteBrickOptions>({
   ctxt: () => ({}),
-  messageContext: (i: number) => ({
-    modComponentId: uuidSequence(i),
-  }),
-  meta: derive<RemoteBrickOptions, RemoteBrickOptions["meta"]>(
-    (options) => ({
-      modComponentId: options.messageContext!.modComponentId,
-      runId: null,
-      branches: [],
-    }),
-    "messageContext",
-  ),
+  meta: runMetadataFactory,
+  messageContext: derive<
+    RemoteBrickOptions,
+    RemoteBrickOptions["messageContext"]
+  >((options) => {
+    assertNotNullish(
+      options.meta,
+      "You must provide meta to derive messageContext",
+    );
+    return mapModComponentRefToMessageContext(options.meta.modComponentRef);
+  }, "meta"),
```

More of a note, but because this is a test, type errors aren't as much of a problem, so we've added an exception in the lint rules allowing `!` in tests. If options isn't defined, the test will fail anyway

We still want strict null checks for tests because it helps call out when we're missing component props, function parameters, etc.

---

For dev ex, I prefer to fail early / provide instructions to the developer. In particular, I think derived fields can be confusing to know which field is the primary field you want to provide